lib/taskana-core/src/main/java/pro/taskana/impl/ClassificationServiceImpl.java

```diff
@@ -323,8 +323,11 @@ public Classification getClassificationByTask(Task task) throws ClassificationNo
             result = classificationMapper.findByTask(classificationKey, task.getWorkbasketKey(),
                 CURRENT_CLASSIFICATIONS_VALID_UNTIL);
             if (result == null) {
+                result = classificationMapper.findByKeyAndDomain(classificationKey, "", CURRENT_CLASSIFICATIONS_VALID_UNTIL);
```

Comment:
Waht´s the difference between byKeyAndDomain() and byTask()? The Query is using a JOIN and we should avoid this. GetByKeyAndDomain() should be finde. Just a single call. 
ClassificationMapper.findByTask can be deleted and shoulöd be deleted.

Reply:
this change should be removed from the PR. getClassificationByTask should be deleted. But this is another story.